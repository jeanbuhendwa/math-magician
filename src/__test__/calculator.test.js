import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/calculator';

describe('Calculator', () => {
  it('should display 0 as default value', () => {
    const { container } = render(<Calculator />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display the correct value when a number button is clicked', () => {
    const { container, getByText } = render(<Calculator />);
    const buttonTwo = getByText('2');
    fireEvent.click(buttonTwo);
    expect(container.firstChild).toMatchSnapshot();
  });
});
