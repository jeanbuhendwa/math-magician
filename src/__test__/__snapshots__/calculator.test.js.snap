// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should display 0 as default value 1`] = `
<main
  class="mainCalcContainer"
>
  <section
    class="imgCalc"
  >
    <img
      alt="Calculate Backgroung"
      src="calculate.svg"
    />
  </section>
  <section
    class="calcContainer"
  >
    <div
      class="displayContainer"
    >
      <h2>
        0
      </h2>
    </div>
    <div
      class="inputContainer"
    >
      <button
        type="button"
      >
        AC
      </button>
      <button
        type="button"
      >
        +/-
      </button>
      <button
        type="button"
      >
        %
      </button>
      <button
        type="button"
      >
        รท
      </button>
      <button
        type="button"
      >
        7
      </button>
      <button
        type="button"
      >
        8
      </button>
      <button
        type="button"
      >
        9
      </button>
      <button
        type="button"
      >
        x
      </button>
      <button
        type="button"
      >
        4
      </button>
      <button
        type="button"
      >
        5
      </button>
      <button
        type="button"
      >
        6
      </button>
      <button
        type="button"
      >
        -
      </button>
      <button
        type="button"
      >
        1
      </button>
      <button
        type="button"
      >
        2
      </button>
      <button
        type="button"
      >
        3
      </button>
      <button
        type="button"
      >
        +
      </button>
      <button
        type="button"
      >
        0
      </button>
      <button
        type="button"
      >
        .
      </button>
      <button
        type="button"
      >
        =
      </button>
    </div>
  </section>
</main>
`;

exports[`Calculator should display the correct value when a number button is clicked 1`] = `
<main
  class="mainCalcContainer"
>
  <section
    class="imgCalc"
  >
    <img
      alt="Calculate Backgroung"
      src="calculate.svg"
    />
  </section>
  <section
    class="calcContainer"
  >
    <div
      class="displayContainer"
    >
      <h2>
        2
      </h2>
    </div>
    <div
      class="inputContainer"
    >
      <button
        type="button"
      >
        AC
      </button>
      <button
        type="button"
      >
        +/-
      </button>
      <button
        type="button"
      >
        %
      </button>
      <button
        type="button"
      >
        รท
      </button>
      <button
        type="button"
      >
        7
      </button>
      <button
        type="button"
      >
        8
      </button>
      <button
        type="button"
      >
        9
      </button>
      <button
        type="button"
      >
        x
      </button>
      <button
        type="button"
      >
        4
      </button>
      <button
        type="button"
      >
        5
      </button>
      <button
        type="button"
      >
        6
      </button>
      <button
        type="button"
      >
        -
      </button>
      <button
        type="button"
      >
        1
      </button>
      <button
        type="button"
      >
        2
      </button>
      <button
        type="button"
      >
        3
      </button>
      <button
        type="button"
      >
        +
      </button>
      <button
        type="button"
      >
        0
      </button>
      <button
        type="button"
      >
        .
      </button>
      <button
        type="button"
      >
        =
      </button>
    </div>
  </section>
</main>
`;
